sdk/core/azure-core/azure/core/pipeline/transport/pyodide.py

```diff
@@ -0,0 +1,160 @@
+            self._content = await self._internal_response.bytes()
+
+    def body(self) -> bytes:
+        """The body is just the content."""
+        return self.content
+
+class PyodideStreamDownloadGenerator(AsyncIterator):
+    """Simple stream download generator that returns the contents of
+    a request.
+    """
+
+    # pylint: disable=unused-argument
+    def __init__(self, pipeline: Pipeline, response: PyodideTransportResponse, *_, **kwargs):
+        self._block_size = response._block_size
+        self.response = response
+        # use this to efficiently store bytes.
+        if kwargs.pop("decompress", False):
+            self._js_reader = response._js_stream.pipeThrough(js.DecompressionStream.new("gzip")).getReader()
+        else:
+            self._js_reader = response._js_stream.getReader()
+        self._stream = BytesIO()
+
+        self._closed = False
+        # We cannot control how many bytes we get from `response.reader`. `self.buffer_left`
+        # indicates how many unread bytes there are in `self.stream`
+        self.buffer_left = 0
+        self.done = False
+
+    async def __anext__(self) -> bytes:
+        """Get the next block of bytes."""
+        if self._closed:
+            raise StopAsyncIteration()
+
+        # remember the initial stream position
+        start_pos = self._stream.tell()
+        # move stream position to the end
+        self._stream.read()
+        # read from reader until there is no more data or we have `self._block_size` unread bytes.
+        while self.buffer_left < self._block_size:
+            read = await self._js_reader.read()
+            if read.done:
+                self._closed = True
+                break
+            self.buffer_left += self._stream.write(bytes(read.value))
+
+        # move the stream position back to where we started
+        self._stream.seek(start_pos)
+        self.buffer_left -= self._block_size
+        return self._stream.read(self._block_size)
+
+class PyodideTransport(AsyncioRequestsTransport):
+    """Implements a basic HTTP sender using the Pyodide Javascript Fetch API.
+
+    WARNING: Pyodide is still an alpha technology. As such, this transport
+    is highly experimental and subject to breaking changes. This transport was
+    built around Pyodide version 0.20.0.
+    """
+
+    async def send(self, request: HttpRequest, **kwargs) -> PyodideTransportResponse:
+        """Send request object according to configuration.
+
+        :param request: The request object to be sent.
+        :type request: ~azure.core.pipeline.transport.HttpRequest
+        :return: An HTTPResponse object.
+        :rtype: PyodideResponseTransport
+        """
+        stream_response = kwargs.pop("stream_response", False)
+        endpoint = request.url
+        request_headers = dict(request.headers)
+        init = {
+            "method": request.method,
+            "headers": request_headers,
+            "body": request.data,
```

Comment:
since i think your transport is `azure.core.rest` only, you should use `content` instead of `data`, since that's the public api for azure.core.rest

Reply:
taked to @iscai-msft offline. Agreed to support old api.